test/unit/exception_handling_test.rb

```diff
@@ -135,11 +148,37 @@ def log_error_callback_with_failure(data, ex)
 
     should "support rescue exceptions from a log_error hook" do
       ExceptionHandling.post_log_error_hook = method(:log_error_callback_with_failure)
+      log_info_messages = []
+      stub(ExceptionHandling.logger).info.with_any_args do |message, _|
+        log_info_messages << message
+      end
       assert_nothing_raised { ExceptionHandling.ensure_safe("mooo") { raise "Some BS" } }
-      assert_equal 0, @fail_count
+      assert log_info_messages.find { |message| message =~ /Unable to execute custom log_error callback/ }
```

Comment:
Since `@fail_count` is only used once in the test, can we move it out of the setup block and into the method that uses it? (line 139)